# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoader.cpp
# Commit: 076b1c7a5a67
# Full Hash: 076b1c7a5a67d4851837a37573be1624d966f921
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2025-09-29 15:42 +0000
# Description:
#   Bug 1991250 - Do not add to the mCacheableDependencyModules twice. r=nbp
# ==============================================================================

diff -r 4e0dfe8abeea -r 076b1c7a5a67 dom/script/ScriptLoader.cpp
--- a/dom/script/ScriptLoader.cpp	Mon Sep 29 15:36:29 2025 +0000
+++ b/dom/script/ScriptLoader.cpp	Mon Sep 29 15:42:04 2025 +0000
@@ -2869,7 +2869,8 @@
        aRequest));
   aRequest->MarkPassedConditionForDiskCache();
 
-  if (aRequest->IsModuleRequest() &&
+  if (!aRequest->PassedConditionForMemoryCache() &&
+      aRequest->IsModuleRequest() &&
       !aRequest->AsModuleRequest()->IsTopLevel()) {
     MOZ_ASSERT(!aRequest->isInList());
     mCacheableDependencyModules.AppendElement(aRequest);
